### ISSUE TITLE
FakeHttpMessageHandler ctor - Wiki update or do I miss something?

### ISSUE THREAD

**Issue by robertmuehsig** [PureKrome/HttpClient.Helpers#14]
I'm using the last recent version 5.0 and tried to follow the Wiki, but it seems the FakeHttpMessageHandler ctor has changed and the wiki is not up to date.

From the wiki I should do something like this:

            var messageHandler = new FakeHttpMessageHandler(requestUrl, messageResponse);

But there is only one parameter:

![image](https://cloud.githubusercontent.com/assets/756703/21429910/38489c9e-c861-11e6-9086-c624326a8d56.png)

I did this in my code and it seems to work:

            var messageResponse = FakeHttpMessageHandler.GetStringHttpResponseMessage(responseData, HttpStatusCode.BadRequest);

            var fakeMessageHandler = new FakeHttpMessageHandler(new HttpMessageOptions() { HttpResponseMessage = messageResponse });

Is this the right approach?

**Comment by PureKrome** [comment]
Hi @robertmuehsig 

yep - that is correct. I bit the bullet and did some breaking changes from 4->5. I was hoping the `semver` bump would be a hint to people that there's been 1+ breaking change applied.

- I more or less nuked the Factory pattern because I felt it was too hard to easy figure out what is going on, if a new person had a look at some code and saw 'magic' happening. I also felt like it was a bit like a service locator pattern. So I decided that DI/IoC was the _safest_ and _most maintainable_ way forward.
- I also replaced all the parameters for DI/IoC with a new `HttpMessageOptions` class. This makes it easier (IMO) to set up your various permutations for endpoint calls.

Secondly, I _thought_ I updated all the wiki items :/ Looks like I didn't at all!

/me goes and updates it.

Ok - all updated.

---

To answer your question directly:

**YES**.

that's the correct way to do it now, if you don't care what URL is passed in.

If you want to make sure that the fake response is returned ONLY if you're calling a specific URL, then change your code to:

```c#
var messageResponse = FakeHttpMessageHandler.GetStringHttpResponseMessage(responseData, HttpStatusCode.BadRequest);

var options = new HttpMessageOptions
{
    RequestUri = "http://www.my.website.com/some/endpoint",
    HttpResponseMessage = messageResponse
};
var fakeMessageHandler = new FakeHttpMessageHandler(options);
```
